virttest/qemu_vm.py

```diff
@@ -3425,6 +3499,63 @@ def hotplug_vcpu(self, cpu_id=None, plug_command="", unplug=""):
         else:
             return(False, cmd_output)
 
+    @error_context.context_aware
+    def hotplug_vcpu_device(self, vcpu_id):
+        """
+        Hotplug a vcpu device and verify that this step is successful.
+        :param vcpu_id: the vcpu id you want to hotplug.
+        """
+        try:
+            vcpu_device = self.devices.get_by_qid(vcpu_id)
+            if vcpu_device[0].get_state():
+                out = "%s is online" % vcpu_id
+                return out, False
```

Comment:
Suggest raise exceptions directly for all the failed cases, and looks we need to introduce the following new exceptions:
```
VMDeviceStateError(VMDeviceError)      # vcpu is already online/offline
VMDeviceInvalidError(VMDeviceError)    # vcpu is not existed
VMDeviceCheckError(VMDeviceError)      # plug/unplug operation failed
```